### PR TITLE
Refactor Code Validation Page UI

### LINKED ISSUE #183: Re-do Test Validations page
The current Test Validations page is based on our old token upload service design with app users reading their codes to contact tracers, and needs to be scrapped and updated to the new design (now available in Figma v3)

https://github.com/google/exposure-notifications-server/blob/master/docs/design/verification_protocol.md

### PR DESCRIPTION
**Pull request by colbymorrison**
Closes #183. Refactor code validation page for v3 with new flow. So far just the UI is redone.

Screen Capture_select-area_20200607233722

### TODO (add these as new issues?)

- Get validation codes from server (is the server set up yet?)

- Display confirmation toast when code is entered on the app (how are we going to do this?)

### CODE REVIEW

File: permission-portal-frontend/client/src/screens/CodeValidations.js [hunk 1]
```diff
@@ -8,70 +7,62 @@ import { withStore } from '../store'
...
+            Provide this validation code to the person you want to verify over the phone
+          </div>
+          <div className="code-box">{code}</div>
+          <PendingOperationButton className="save-button generate-button" operation={genNewCode}>
+            Generate New Code
+          </PendingOperationButton>
         </div>
       </div>
-      <Toast ref={confirmedToast} isSuccess={true} message="Code verification confirmed" />
+      <Toast ref={confirmedToast} isSuccess={toastInfo.success} message={toastInfo.msg} />
```
Comment (fordacious) [inline permission-portal-frontend/client/src/screens/CodeValidations.js:65]: I think ref is an antipattern in react and discouraged. In other pages we've been using state.

File: permission-portal-frontend/client/src/screens/CodeValidations.js [hunk 2]
```diff
@@ -8,70 +7,62 @@ import { withStore } from '../store'
...
+            <h2>Validation Code</h2>
             <div className="tooltip">
               <img src={info_icon} alt="info" />
               <div className="tooltip-msg">
-                {/* to replace tooltip text here: */}
-                <div className="tooltip-title">This the tooltip title</div>
+                {/* TODO replace tooltip text */}
+                <div className="tooltip-title">Test Validation Codes</div>
                 <div className="tooltip-body">
                   Quisque sagittis, vel hendrerit consectetur tincidunt senectus. Feugiat aenean nunc, tempus tempus,
```
Comment (fordacious) [inline permission-portal-frontend/client/src/screens/CodeValidations.js:50]: nit: I'd put a TODO in here so it is not missed (easier to search for things that are not done)

File: permission-portal-frontend/client/src/screens/CodeValidations.js [hunk 3]
```diff
@@ -8,70 +7,62 @@ import { withStore } from '../store'
...
+    msg: '',
+  })
 
   let confirmedToast = createRef()
 
+  //TODO show confired toast when code confirmed by app
+
+  const genNewCode = async () => {
+    // TODO get new code from server
+    try {
```
Comment (fordacious) [inline permission-portal-frontend/client/src/screens/CodeValidations.js:28]: can use promises without try catch

File: permission-portal-frontend/client/src/screens/CodeValidations.js (hunk 1 above)
Comment (colbymorrison) [reply permission-portal-frontend/client/src/screens/CodeValidations.js:65]: After [PR 193](https://github.com/covid19risk/permission-portal/pull/193) this is the way all the toasts work across the app, so I was just following those.

File: permission-portal-frontend/client/src/screens/CodeValidations.js (hunk 2 above)
Comment (colbymorrison) [reply permission-portal-frontend/client/src/screens/CodeValidations.js:50]: I believe that's what I did with the changes at lines 47-48. But let me know if I misinterpreted what you meant.

File: permission-portal-frontend/client/src/screens/CodeValidations.js (hunk 1 above)
Comment (aidan-fitz) [reply permission-portal-frontend/client/src/screens/CodeValidations.js:65]: I changed all the toasts to refs. Happy to discuss alternatives.

Comment (fordacious) [reply permission-portal-frontend/client/src/screens/CodeValidations.js:65]: I'd say I'm fine if it's consistent and works without issue.

File: permission-portal-frontend/client/src/screens/CodeValidations.js (hunk 3 above)
Comment (colbymorrison) [reply permission-portal-frontend/client/src/screens/CodeValidations.js:28]: I'll add this when we know how we're going to talk to the server. Right now this method is just a placeholder

**Comment by ibeckermayer** [comment]
> * Get validation codes from server (is the server set up yet?)

No, the server is not set up yet. [Issue](https://github.com/covid19risk/permission-portal/issues/250)

> * Display confirmation toast when code is entered on the app (how are we going to do this?)

Not sure about this, I asked in Figma, hold off on making any issues.

Also added [an issue](https://github.com/covid19risk/permission-portal/issues/251) for updating the lorem ipsum popout (blocked).

**Pull request merged**